src/sync/sync_config.hpp

```diff
@@ -105,6 +105,21 @@ struct SyncConfig {
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
     std::shared_ptr<ChangesetTransformer> transformer;
+#if __GNUC__ < 5
+// GCC 4.9 does not support C++14 braced-init with NSDMIs
```

Comment:
I'm a little confused by this as there are no NSDMIs in this class. Why is it really necessary?